merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -235,14 +256,20 @@ export class Discrete {
     })
   }
 
-  public equal(other: Discrete): Windows {
+  public equal(other: any): Windows {
```

Comment:
Matt, Adrien, and I came up with a clever solution for constraining these types in a huddle this morning. I'm sure Matt will be working with you on getting that put in here soon enough.

Reply:
In this particular instance, the type is intentionally not constrained. It allows you to use `discreteResource.equal("A string")` instead of `discreteResource.equal(Discrete.Value("A string"))`, because I discovered that constantly wrapping all literals in `Discrete.Value` or `Real.Value` got very tedious, very fast. In the implementation for the function, it checks if the input is already `instanceof Discrete`, and if not will wrap it in `Discrete.Value`.

Reply:
The reason for the change from `Discrete` to `Any` is to avoid wrapping values in `Discrete.Value(...)` when its redundant. If we could have a `Discrete<T>` type, then I would do the following:

```typescript
public equal(other: Discrete<T> | T): Windows {
    if (other instanceof T) {
        other = Discrete.Value(other);
    }
    // carry on
}
```

The problem is determining the type `T` when the left-hand operand is created from `Discrete.Resource` or `Discrete.Parameter`. I understand how I'd type-check the resource names, like this:

```typescript
public enum Resource {
    MyResource = "My Resource",
    Resource2 = "Resource Two"
}

// ... in the Discrete class

public static Resource(resource: Resource): Discrete {
    // do the thing
}
```

How would I transform that function into returning `Discrete<T>`?

Reply:
marking resolved, we can talk about this as part of the next PR.